src/System.Management.Automation/engine/parser/PSType.cs

```diff
@@ -1118,13 +1118,17 @@ internal static Assembly DefineTypes(Parser parser, Ast rootAst, TypeDefinitionA
                                       : rootAst.Extent.File
                                       .Replace('\\', (char)0x29f9)
                                       .Replace('/', (char)0x29f9)
-                                      .Replace(',', (char)0x201a)
                                       .Replace(':', (char)0x0589));
 
-            var assembly = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName(assemblyName),
+            // We can not use 'AssemblyName(string assemblyName)' constructor because it is trying to parse the assembly name
+            // as 'Name <,Culture = CultureInfo> <,Version = Major.Minor.Build.Revision> <, StrongName> <,PublicKeyToken> '\0''
+            // See notes in https://msdn.microsoft.com/en-us/library/system.reflection.assemblyname(v=vs.110).aspx
+            var assemblyObj = new AssemblyName();
+            assemblyObj.Name = assemblyName;
```

Comment:
Here is my approach - it could use a test but it works: https://github.com/lzybkr/PowerShell/tree/fix_dyn_class_assem